File: API/src/main/java/com/ltizzi/ecommerce/model/stockEntry/StockEntryEntity.java
package com.ltizzi.ecommerce.model.stockEntry;

import com.ltizzi.ecommerce.model.product.ProductEntity;
import com.ltizzi.ecommerce.model.stock.StockEntity;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.Where;

import java.sql.Timestamp;

/**
 * @author Leonardo Terlizzi
 */

@Data
@Entity
@Table(name = "stock-entries")
@SQLDelete(sql = "UPDATE stock-entries SET soft_delete = true where entrie_id =?")
@Where(clause = "soft_delete = false")
public class StockEntryEntity {

    @Id
    @GeneratedValue (strategy = GenerationType.AUTO)
    private Long entry_id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    private ProductEntity product;

    @Column(nullable = false)
    private int cantidad;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name= "stock_id")
    private StockEntity stock;

    @CreationTimestamp
    @Column(updatable = false)
    private Timestamp createdAt;

    @UpdateTimestamp
    private Timestamp updatedAt;

    @Column(name="soft_delete")
    private Boolean soft_delete = Boolean.FALSE;
}

File: Server/src/entities/StockEntryEntity.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  OneToOne,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
} from "typeorm";
import { Product } from "../models/Product";
import { Stock } from "../models/Stock";
import { StockEntry } from "../models/StockEntry";
import { ProductEntity } from "./ProductEntity";
import { StockEntity } from "./StockEntity";

@Entity("stock_entries")
export class StockEntryEntity implements StockEntry {
  @PrimaryGeneratedColumn()
  entry_id!: number;

  @ManyToOne(() => ProductEntity, { eager: false, nullable: true })
  @JoinColumn({ name: "product_id", referencedColumnName: "product_id" })
  product!: Product;

  @Column()
  cantidad!: number;

  @Column()
  stock_id!: number;

  @ManyToOne(() => StockEntity, { eager: true, nullable: false })
  @JoinColumn({ name: "stock", referencedColumnName: "stock_id" })
  stock!: Stock;

  @CreateDateColumn({
    type: "timestamp",
    default: () => "CURRENT_TIMESTAMP",
  })
  createdAt!: Date;

  @UpdateDateColumn({
    type: "timestamp",
    default: () => "CURRENT_TIMESTAMP",
    onUpdate: "CURRENT_TIMESTAMP",
  })
  updatedAt!: Date;

  @Column({ default: false })
  soft_delete!: boolean;
}
